README.md

```diff
@@ -1,2 +1,270 @@
+
+This document describes how data is stored in Firestore for the Engage-HF app.
+
+## Custom Types
+
+The following section describes custom types defined for this system to be later used
+
+### LocalizedText
+
+A LocalizedText object shall be used whenever text requires localization or may require in the future. The object may either simply be a single string (in the case of localizations not being available yet) or a dictionary with string keys and string values. If LocalizedText is represented by a string-string dictionary, the keys represent language-codes following the ISO 639-1 (or if necessary ISO 639-2) standard and the respective text in the given language as value.
+
+Some localizations may require to include regions as well (e.g. Australian/British/American English), so the clients should be aware of this and prioritize the language code with the correct region (`en-us`) over the general code (`en`), if it is present.
+
+A LocalizedText object cannot be used in FHIR-conforming types due to its incompatibility with the standard. FHIR types commonly contain text in one language only to be specified using the `language` property.
+
+#### Example
+
+```
+{
+	"en":"Welcome to the Engage-HF app!",
+	"de":"Willkommen in der Engage-HF App!",
+	"es":"¡Bienvenido a la aplicación Engage-HF!"
+}
+```
+
+
+## /questionnaires
+
+In this section, we describe all the information stored for questionnaires.
+
+### /questionnaires/$questionnaireId$
+
+Based on [FHIR Questionnaire](https://hl7.org/fhir/R4B/questionnaire.html):
+
+|Property|Type|Values|Comments|
+|-|-|-|-|
+|id|string|-|[Resource](https://hl7.org/fhir/R4B/resource.html): Logical id of this artifact|
+|meta|Meta|-|[Resource](https://hl7.org/fhir/R4B/resource.html#Meta): Metadata about the resource|
+|identifier|list of Identifier|-|Business identifier for this questionnaire. There may be multiple questionnaire objects in this list, with differing `id` and `language` properties, but a common business identifier.|
+|title|string|-|Human-friendly title of the questionnaire.|
+|date|DateTime|-|last modified date|
+|version|string|-|Business version of the questionnaire.|
+|language|optional string|e.g. "en"|Following IETF BCP-47 / [FHIR ValueSet languages](https://hl7.org/fhir/R4B/valueset-languages.html)|
+|item|list of BackboneElement|-|items as defined in [FHIR Questionnaire](https://hl7.org/fhir/R4B/questionnaire.html)|
+|item[x]>linkId|string|-|Unique id for item in questionnaire|
+|item[x]>definition|optional uri|-|Details for the item|
+|item[x]>prefix|optional string|e.g. "1(a)", "2.5.3"|Prefix of the item that isn't actually part of the name but may still be displayed.|
+|item[x]>text|string|-|Primary text for the item|
+|item[x]>type|code|e.g. "group", "display", "boolean", "decimal", "integer", "date", etc|See [QuestionnaireItemType](https://hl7.org/fhir/R4B/valueset-item-type.html) for available values.|
+
+#### Example
+
+TODO: Create KCCQ-12 questionnaire using FHIR data format.
```

Comment:
That would be great; I think we should have all the information in the UI document. 

You can use our [phoenix](https://stanfordbdhg.github.io/phoenix/) project for this.

Might make sense to add this as a separate file in the repo here, would be nice to include this in this PR.

Reply:
Thank you, will do!